### DISCUSSION TITLE
Unable to flash Freetz-NG on F!B 6660 v7.57

### DISCUSSION THREAD

**Discussion by VirusBurst** [Freetz-NG/freetz-ng#888] [push firmware]
Hello there :)

I've recently flashed Freetz-NG on an F!B 6490 with the firmware 7.57 => works fine (hardware series 7490/7590, hardware type 6490)
Now I wanted to flash Freetz-NG on my F!B 6660 with the only difference => hardware type to 6660

Make was successful, power-cycle the F!B opened the bootloader for flashing which doesn't give any errors but then... nothing!

F!B 6660 doesn't do anything, no lights, nothing... After a power-cycle, it reboots into the OFW like nothing was ever flashed onto it.
[6660v757.zip](https://github.com/Freetz-NG/freetz-ng/files/13328867/6660v757.zip)

I've added the logs of push_firmware and my .config file, maybe you find the issue I've overseen.
I also tried to make and flash the image with hardware series set to "All devices" but that didn't help either :/

Hope you can help me ♥

**Comment by Conan179** [comment]
How do you flash the image?

**Reply by VirusBurst** [reply]
Oh sorry, I forgot;

I build the images with debian on a notebook. I also tried one from the Ubuntu Images via VM and with Kali-Linux via WSL on Windows. 
Flashing only with the debian notebook!

**Reply by Conan179** [reply]
um yes, how do you flash the image EXACTLY?

**Reply by VirusBurst** [reply]
Just 'tools/push_firmware' inside the terminal and the script ask me, whether image XYZ is the correct one. 
Although... I don't remember whether I set 'umask 0022' when I flashed the F!B 6490 image... I had to do that twice because before, the OFW wasn't updated to 7.57 so maybe I set 'umask 0022' before flashing...

**Comment by fda77** [comment]
Some AVM devices toggle the "linux_fs_start" if an image crashes or does not boot. You could check by the bootloader if it changed

**Reply by VirusBurst** [reply]
Hmmm... this are the last lines of the flash script

---> SETENV linux_fs_start 1
200 SETENV command successful
---> REBOOT
221 Thank you for using the FTP service on ADAM2
---> QUIT
221 Goodbye.

For me, it looks like it get toggled to 1 to boot from the image but the image is faulty (or flashed wrong) and after a power-cycle, it reuses the other partition with the working OFW

**Reply by VirusBurst** [reply]
I have BIOS EDD facility v0.16 2004-Jun-25... and I think... you can only flash via UART... 

More bad news ... somehow, I've managed to buy an "unbranded" Vodafone Router... 20002932... and they didn't noticed yet LOL

**Reply by fda77** [reply]
2004? I dont think these devices are so old
https://bitbucket.org/fesc2000/ffritz/src/6591/README-6591.md
Maybe you should start with a minimal-image. But i cant say how stable Puma7 devices are, i never had one

**Comment by VirusBurst** [comment]
> 2004? I dont think these devices are so old https://bitbucket.org/fesc2000/ffritz/src/6591/README-6591.md Maybe you should start with a minimal-image. But i cant say how stable Puma7 devices are, i never had one

Hello there, I'm still here ^^

TODAY I finally flashed Freetz-NG... I used an image from Prisrak called "6660_07.57.ger_freetz-ng-23469MOA.image"

but it was a very very FING tough night!

But I still want to discuss something...

First of all... although I own the Fritzbox Cable 6660... and Freetz-NG always made/identified the images as "ProductID Fritz_Box_HW252a" MY 6660 Cable identifies as "ProductID Fritz_Box_HW252a" and therefore... the push_firmware (somehow) always refused to flash the images... 

I had to insert a break after reading the env-file to put the "a" to the ProductID so the flasher can do it's job...
Anyways the issues didn't stop here...

It seems... somehow... my Fritzbox refuse to boot off the second slot... it always revert back to "linux_fs_start = 0" ... until I asked myself... what if I force the flasher to use the active slot via "lfs = 0" and TA-DA... I am back... 

So... yeah... now, what can I do, to investigate why my own made images failed booting... ?

**Reply by fda77** [reply]
Dieser Test der ProductID ist erst 5 Tage alt und sollte damit weder funktionieren: 

* c5edfbb62 push_firmware: fix HWR for multi cpu devices

Manche Geräte haben mehrere CPUs, bei dir ist das ein ARM und ein Atom, und avm hat jedem eine eigene HWR gegeben

Ansonsten wäre die komplette Ausgabe von PF sinvoll!!

Nimm doch die .config aus dem Image und bau damit mal eine eigenes Image
